Tests/OTelCoreTests/Configuration/OTelConfigurationTests.swift

```diff
@@ -0,0 +1,763 @@
+    // https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/
+    @Test(.disabled("Not currently implemented")) func testSDKDisabled() {}
+
+    // OTEL_LOG_LEVEL
+    // https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/
+    @Test func testDiagnosticsLogLevel() {
+        #expect(OTel.Configuration.default.diagnosticLogLevel.backing == .info)
+
+        #expect(OTel.Configuration.default.applyingEnvironmentOverrides(environment: [
+            "OTEL_LOG_LEVEL": "trace",
+        ]).diagnosticLogLevel.backing == .trace)
+    }
+
+    // OTEL_SERVICE_NAME
+    // https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/
+    // https://opentelemetry.io/docs/languages/sdk-configuration/general/#otel_service_name
+    @Test func testServiceName() {
+        #expect(OTel.Configuration.default.serviceName == "unknown_service")
+
+        #expect(OTel.Configuration.default.applyingEnvironmentOverrides(environment: [
+            "OTEL_SERVICE_NAME": "some_service",
+        ]).serviceName == "some_service")
+
+        withKnownIssue {
+            Issue.record("""
+            This property is a special since it's supposed to affect what ends up as the value for the service.name
+            resource attribute during export.
+
+            However, I don't think the right way to do that is to magically change the resource attributes of the confg
+            object that the user has.
+
+            Most likely this should be solved with a computed property on the config that's used by exporters, e.g.
+            `resolvedResourceAttributes`.
+            """)
+        }
+
+        #expect(OTel.Configuration.default.applyingEnvironmentOverrides(environment: [
+            "OTEL_RESOURCE_ATTRIBUTES": "service.name=some_service_from_resource_attributes",
+            "OTEL_SERVICE_NAME": "some_service_from_service_name",
+        ]).serviceName == "some_service_from_service_name")
+    }
+
+    // OTEL_RESOURCE_ATTRIBUTES
+    // https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/
+    // https://opentelemetry.io/docs/languages/sdk-configuration/general/#otel_resource_attributes
+    // https://opentelemetry.io/docs/specs/otel/resource/sdk/#specifying-resource-information-via-an-environment-variable
+    @Test func testResourceAttributes() {
+        #expect(OTel.Configuration.default.resourceAttributes.isEmpty)
+
+        #expect(OTel.Configuration.default.applyingEnvironmentOverrides(environment: [
+            "OTEL_RESOURCE_ATTRIBUTES": "key1=value1,key2=value2",
+        ]).resourceAttributes == ["key1": "value1", "key2": "value2"])
+
+        #expect(OTel.Configuration.default.applyingEnvironmentOverrides(environment: [
+            "OTEL_RESOURCE_ATTRIBUTES": "key=first_value,key=second_value",
+        ]).resourceAttributes == ["key": "second_value"])
+
+        // This is a special key that needs merging (most keys just overwrite).
```

Comment:
It took me quite a while to understand this comment. I think what it's trying to say is that in this part of the test we're testing that env and code keys are being merged and that env values override code values? Perhaps writing it like that would be a bit clearer. I thought at first you meant something along the lines of "only this special key is being overridden, otherwise we don't override anything".

Reply:
That's fair. I've expanded this comment a bunch in a fixup commit.

It now reads:

```
// The resource attributes provided as an environment variable are a bit special. Most configuration values are
// taken from the environment variable as-is, and completely override the configuration value, but for resource
// attributes, the values from the environment variable are considered additional and should be merged into the
// existing value.
```

Hopefully that's clear enough.